### PR TITLE
[baremetal] ccm - clean temp data

### PR DESCRIPTION
**Pull request by AndrzejKurek**
This PR adds buffer clearing on top of https://github.com/ARMmbed/mbedtls/pull/3840.

### CODE REVIEW

**Comment by shelib01** [comment]
I tried to re-run pr-merge, but didn't succeed . Can be merged after tests pass

**Review by shelib01 — DISMISSED** [review]

**Comment by AndrzejKurek** [comment]
Had to rebase the PR. Waiting for CI.

**Comment by AndrzejKurek** [comment]
Tests passed on this one.

**Review by shelib01 — APPROVED** [review]

**Pull request merged**
